clippy_lints/src/float_literal.rs

```diff
@@ -82,20 +82,20 @@ impl<'tcx> LateLintPass<'tcx> for FloatLiteral {
                     LitFloatType::Suffixed(ast::FloatTy::F64) => Some("f64"),
                     LitFloatType::Unsuffixed => None
                 };
-                let (is_whole, mut float_str) = match fty {
+                let (is_whole, is_inf, mut float_str) = match fty {
                     FloatTy::F32 => {
                         let value = sym_str.parse::<f32>().unwrap();
 
-                        (value.fract() == 0.0, formatter.format(value))
+                        (value.fract() == 0.0, value.is_infinite(), formatter.format(value))
                     },
                     FloatTy::F64 => {
                         let value = sym_str.parse::<f64>().unwrap();
 
-                        (value.fract() == 0.0, formatter.format(value))
+                        (value.fract() == 0.0, value.is_infinite(), formatter.format(value))
                     },
```

Comment:
nit: I think this is cleaner, personally:

```rust
let value = if FloatTy::F32 {
	sym_str.parse::<f32>().unwrap()
} else {
	sym_str.parse::<f64>().unwrap()
};

let (is_whole, is_inf, mut float_str) = (value.fract() == 0, value.is_infinite(), formatter.format(value));
```

Reply:
That can't work unfortunately, due to mismatched types